### PR TITLE
Better handle undefined Error stacks in DevTools error boundary

### LINKED ISSUE #23373: [DevTools Bug] Cannot read properties of undefined (reading 'split')
### Website or app

https://next-rsc-notes.vercel.app/

### Repro steps

1. enter the site
2. open react-devtools
3. select We(maybe suspense's child component)
4. and show following errors.

### How often does this bug happen?

Every time

### DevTools package (automated)

react-devtools-extensions

### DevTools version (automated)

4.23.0-e28a0db22

### Error message (automated)

Cannot read properties of undefined (reading 'split')

### Error call stack (automated)

```text
at getDerivedStateFromError (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37414:114)
    at ErrorBoundary_ErrorBoundary.c.payload (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:16143:14)
    at Gg (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:14571:47)
    at Dj (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:16788:5)
    at jl (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:19204:86)
    at il (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:18756:11)
    at hl (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:18748:23)
    at Wk (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:18732:5)
    at al (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:18393:37)
    at Uk (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:18314:51)
```

### Error component stack (automated)

```text
at ErrorBoundary_ErrorBoundary (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37375:5)
    at div
    at InspectedElementErrorBoundaryWrapper (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37864:3)
    at NativeStyleContextController (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:40513:3)
    at div
    at div
    at OwnersListContextController (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:33354:3)
    at SettingsModalContextController (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:35975:3)
    at Components_Components (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:42520:52)
    at ErrorBoundary_ErrorBoundary (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37375:5)
    at div
    at div
    at ThemeProvider (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37518:3)
    at PortaledContent (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37548:5)
    at div
    at div
    at div
    at ThemeProvider (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37518:3)
    at TimelineContextController (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:42700:3)
    at ProfilerContextController (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:42146:3)
    at TreeContextController (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:30256:3)
    at SettingsContextController (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:30878:3)
    at ModalDialogContextController (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:37927:3)
    at DevTools_DevTools (chrome-extension://fmkadmapgofadopljbjfkapdkoienihi/build/main.js:53807:3)
```

### GitHub query string (automated)

```text
https://api.github.com/search/issues?q=Cannot read properties of undefined (reading 'split') in:title is:issue is:open is:public label:"Component: Developer Tools" repo:facebook/react
```

### PR DESCRIPTION
**Pull request by bvaughn**
Resolves #23373

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ErrorBoundary.js [hunk 1]
```diff
@@ -49,16 +49,16 @@ export default class ErrorBoundary extends Component<Props, State> {
     const errorMessage =
       typeof error === 'object' &&
       error !== null &&
-      error.hasOwnProperty('message')
+      typeof error.message === 'string'
         ? error.message
-        : String(error);
+        : null;
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ErrorBoundary.js:54]: This change prevents us from searching for a matching GitHub issue when we have no unique error message to search with.

File: packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ErrorBoundary.js [hunk 2]
```diff
@@ -49,16 +49,16 @@ export default class ErrorBoundary extends Component<Props, State> {
...
 
     const isTimeout = error instanceof TimeoutError;
 
     const callStack =
       typeof error === 'object' &&
       error !== null &&
-      error.hasOwnProperty('stack')
+      typeof error.stack === 'string'
```
Comment (bvaughn) [inline packages/react-devtools-shared/src/devtools/views/ErrorBoundary/ErrorBoundary.js:61]: This change prevents us from trying to call `String.prototype.split` on an undefined or null value.

**Comment by lunaruan** [comment]
Looks good! Could you take a screenshot?

**Review by lunaruan — APPROVED** [review]

**Comment by bvaughn** [comment]
Screen Shot 2022-03-10 at 1 36 24 PM

**Pull request merged**